project/BouncyCastlePlugin.scala

```diff
@@ -0,0 +1,50 @@
+import com.typesafe.tools.mima.plugin.MimaKeys.*
+import org.typelevel.sbt.NoPublishPlugin
+import sbt.*
+import sbt.Keys.*
+
+case class ArtifactVersions(latest: ModuleID, compatibleVersions: Seq[String]) {
+  val history: Set[ModuleID] = compatibleVersions.map(latest.withRevision).toSet
+}
+
+object BouncyCastlePlugin extends AutoPlugin {
+
+  override def trigger = allRequirements
+
+  object autoImport {
+    lazy val bouncyCastleArtifacts: Seq[ModuleID] = modules.map(_.latest)
+  }
+
+  /* When a new BouncyCastle version is released, add it to the beginning of
+   * these lists. If `rootBouncyCastle/mimaReportBinaryIssues` shows no issues,
+   * we have a chain of binary-compatible releases and the new list can
+   * be left as it is (i.e. with the new version at the head). If issues
+   * are found, then remove everything other than the new version,
+   * because the newest release is not binary compatible with the older
+   * versions it was checked against.
+   */
+  lazy val modules = Seq(
+    ArtifactVersions("org.bouncycastle" % "bcpg-jdk18on" % "1.72.2", Seq("1.72.1")),
```

Comment:
Hopefully Scala Steward is smart enough to update the primary versions here. I'm sure it won't add the previous version to the `compatibleVersions` list, but if it at least creates the PR we can take it from there.